docker-compose.yaml

```diff
@@ -19,79 +19,126 @@ x-default: &node
     - GH_PEM
     - IPTS_PRD_PASSWORD
     - IPTS_TST_PASSWORD
-  networks:
-    - sparknet
+
+# These environmental variables get templated/merged into the env vars of
+# containers. Usually YAML doesn't let you merge arrays, but Compose allows it:
+# https://docs.docker.com/reference/compose-file/extension/#example-4
+x-spark-env: &spark-env
+  SPARK_MASTER_URL: spark://spark-node-master:7077
+  SPARK_PUBLIC_DNS: ${SPARK_PUBLIC_IP}
+  SPARK_LOCAL_STORAGE_ENCRYPTION_ENABLED: no
+  SPARK_RPC_AUTHENTICATION_ENABLED: no
+  SPARK_RPC_ENCRYPTION_ENABLED: no
+  SPARK_SSL_ENABLED: no
+
+  AWS_SHARED_CREDENTIALS_FILE: /run/secrets/AWS_CREDENTIALS
+  AWS_S3_BUCKET:
+  AWS_S3_PREFIX:
+  GH_APP_ID:
+
+  IPTS_PRD_SERVICE_NAME:
+  IPTS_PRD_HOSTNAME:
+  IPTS_PRD_USERNAME:
+  IPTS_PRD_PORT:
+  IPTS_TST_SERVICE_NAME:
+  IPTS_TST_HOSTNAME:
+  IPTS_TST_USERNAME:
+  IPTS_TST_PORT:
 
 services:
-  spark-node-master:
+  spark-node-master-prod:
     <<: *node
-    container_name: spark-node-master
+    profiles: ["prod", ""] # empty string here runs when no profile is specified
+    image: ${CCAO_REGISTRY_URL}/service-spark-iasworld:latest
+    container_name: spark-node-master-prod
     hostname: spark-node-master
+    restart: unless-stopped
     environment:
-      - SPARK_MODE=master
-
-      - AWS_SHARED_CREDENTIALS_FILE=/run/secrets/AWS_CREDENTIALS
-      - AWS_S3_BUCKET
-      - AWS_S3_PREFIX
-      - SPARK_LOCAL_STORAGE_ENCRYPTION_ENABLED=no
-      - SPARK_PUBLIC_DNS=${SPARK_PUBLIC_IP}
-      - SPARK_RPC_AUTHENTICATION_ENABLED=no
-      - SPARK_RPC_ENCRYPTION_ENABLED=no
-      - SPARK_SSL_ENABLED=no
-      - SPARK_USER=spark
-      - IPTS_PRD_SERVICE_NAME
-      - IPTS_PRD_HOSTNAME
-      - IPTS_PRD_USERNAME
-      - IPTS_PRD_PORT
-      - IPTS_TST_SERVICE_NAME
-      - IPTS_TST_HOSTNAME
-      - IPTS_TST_USERNAME
-      - IPTS_TST_PORT
-      - GH_APP_ID
+      <<: [*spark-env]
```

Comment:
This bizarre syntax seems specific to Compose and lets you merge maps i.e. combine all the env vars above with the ones specified here.

Reply:
We don't! It was just left over from when I was merging multiple anchors (I'd split the env vars up into groups).